File: android/DFM/src/main/java/com/dontflymoney/layout/AccountLine.java
package com.dontflymoney.layout;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.dontflymoney.view.ExtractActivity;
import com.dontflymoney.view.R;

import org.json.JSONException;

import java.text.DecimalFormat;

public class AccountLine extends LinearLayout
{
	public AccountLine(Context context, AttributeSet attributeSet)
	{
		super(context, attributeSet);
	}

	@Override
	protected void onFinishInflate()
	{
		super.onFinishInflate();
		NameField = (TextView)findViewById(R.id.name);
		TotalField = (TextView)findViewById(R.id.value);
	}

	public TextView NameField;
	public TextView TotalField;

	public void setAccount(AccountAdapter.Account account, int color) throws JSONException
	{
		setBackgroundColor(color);

		NameField.setText(account.Name);

		int totalColor = account.Total < 0 ? Color.RED : Color.BLUE;
		double totalToShow = account.Total < 0 ? -account.Total : account.Total;
		String totalStr = new DecimalFormat("#,##0.00").format(totalToShow);

		TotalField.setTextColor(totalColor);
		TotalField.setText(totalStr);

		setClickable(true);

		setOnClickListener(new onClickListener(getContext(), account.Url));
	}

	public class onClickListener implements OnClickListener
	{
		private Context context;
		private String url;

		onClickListener(Context context, String url)
		{
			this.context = context;
			this.url = url;
		}

		@Override
		public void onClick(View v)
		{
			Intent intent = new Intent(context, ExtractActivity.class);
			intent.putExtra("accounturl", url);
			context.startActivity(intent);
		}
	}

}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/uiHelpers/views/AccountLine.kt
package com.darakeon.dfm.uiHelpers.views

import android.content.Context
import android.content.Intent
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import com.darakeon.dfm.R
import com.darakeon.dfm.activities.ExtractActivity
import com.darakeon.dfm.activities.base.setColorByAttr
import com.darakeon.dfm.uiHelpers.adapters.AccountAdapter
import kotlinx.android.synthetic.main.accounts_line.view.*
import java.text.DecimalFormat

class AccountLine(context: Context, attributeSet: AttributeSet) : LinearLayout(context, attributeSet) {

	fun setAccount(account: AccountAdapter.Account, color: Int) {
		setBackgroundColor(color)

		name.text = account.name

		val totalColor = if (account.total < 0) R.attr.negative else R.attr.positive
		val totalToShow = if (account.total < 0) -account.total else account.total
		val totalStr = DecimalFormat("#,##0.00").format(totalToShow)

		value.setColorByAttr(totalColor)
		value.text = totalStr

		isClickable = true

		setOnClickListener(OnClickListener(context, account.url))
	}

	inner class OnClickListener internal constructor(private val context: Context, private val url: String) : View.OnClickListener {

		override fun onClick(v: View) {
			val intent = Intent(context, ExtractActivity::class.java)
			intent.putExtra("accountUrl", url)
			context.startActivity(intent)
		}
	}
}
